tests/bluetooth/bsim_bt/bsim_test_eatt/src/main_collision.c

```diff
@@ -0,0 +1,227 @@
+/* main_collision.c - Application main entry point */
+
+/*
+ * Copyright (c) 2022 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stddef.h>
+
+#include <zephyr/types.h>
+#include <sys/printk.h>
+#include <sys/util.h>
+#include <sys/byteorder.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/att.h>
+
+#include "bs_types.h"
+#include "bs_tracing.h"
+#include "bstests.h"
+
+extern enum bst_result_t bst_result;
+
+#define FAIL(...)                                                                                  \
+	do {                                                                                       \
+		bst_result = Failed;                                                               \
+		bs_trace_error_time_line(__VA_ARGS__);                                             \
+	} while (0)
+
+#define PASS(...)                                                                                  \
+	do {                                                                                       \
+		bst_result = Passed;                                                               \
+		bs_trace_info_time(1, __VA_ARGS__);                                                \
+	} while (0)
+
+static struct bt_conn *default_conn;
+
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+};
+
+static bool volatile is_connected;
+
+static void connected(struct bt_conn *conn, uint8_t conn_err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (conn_err) {
+		if (default_conn) {
+			bt_conn_unref(default_conn);
+			default_conn = NULL;
+		}
+
+		FAIL("Failed to connect to %s (%u)\n", addr, conn_err);
```

Comment:
Isn't there a missing return statement here?

Reply:
When a `FAIL()` statement is reached, the test is aborted.

Reply:
But the callback function is not returned from (at least looking at the FAIL macro definition). So it still seems to me like the correct thing to do to add the return statement there. Otherwise the code needlessly calls functions like `bt_conn_ref()` and sets `is_connected` to `true`.

Reply:
IIRC, the `FAIL` macro here will set `bst_result = Failed`, and on the next tick of the simulator it will end the case. 

The `return` is to my understanding not needed (although perhaps nice to have for semantic reasons. @aescolar What is your take?

Reply:
At the very least there should be a code comment (but I'd still prefer a return statement, even though it doesn't make any practical difference), since for someone not familiar with the simulator it looks very much like a bug.

Reply:
Will add the return.